client/src/components/ContestDetail.js

```diff
@@ -105,26 +113,43 @@ const submitLink = React.forwardRef((props, ref) => {
   return <Link innerRef={ref} to={`${id}/submit`} {...props} />;
 });
 
-const ContestDetail = ({ classes, auth, contest, match, history }) => {
+const ContestDetail = ({ classes, auth, match, history }) => {
   const [tabPage, setTabPage] = useState(0);
   const [sub, setSub] = useState(null);
-  const { title, prize, creator, submissions } = contest;
-  const isCreator = contest.creator._id === auth.user.userId;
   const [open, setOpen] = useState(false);
+  const [contest, setContest] = useState(null);
```

Comment:
while moving the fetch down will work, I really did it like this to ensure the separation of fetching component and display. A better solution would've been to just move down setContest and then call it whenever you make a change e.g.  selecting a winner